src/templates/doubleKeyedMap.h

```diff
@@ -156,4 +162,33 @@ template <typename ValueClass> class DoubleKeyedMap
             { result[{i, j}] = find(keyGetter(itemI), keyGetter(itemJ)); }, mirroredAreEquivalent_);
         return result;
     }
+
+    /*
+     * Serialisation
+     */
+    public:
+    // Express as a serialisable value
+    SerialisedValue serialise() const override
+    {
+        SerialisedValue result;
+        Serialisable::fromMap(data_, "map", result);
+        return result;
+    };
+    // Read values from a serialisable value
+    void deserialise(const SerialisedValue &node)
+    {
+        data_.clear();
+
+        for (auto &[mapKey, value] : toml::find<SerialisedValue::table_type>(node, "map"))
+        {
+            if constexpr (std::is_same_v<ValueClass, double>)
+                data_[mapKey] = value.as_floating();
+            else if constexpr (std::is_same_v<ValueClass, int>)
+                data_[mapKey] = value.as_integer();
+            else if constexpr (std::is_same_v<ValueClass, bool>)
+                data_[mapKey] = value.as_boolean();
+            else
+                data_[mapKey].deserialise(value);
+        }
```

Comment:
We could use `toMap` here for symmetry, but I'm not sure that we get that much out of it.

```suggestion
        toMap(node["map"], [this](const auto &mapKey, const auto &value)
        {
            if constexpr (std::is_same_v<ValueClass, double>)
                data_[mapKey] = value.as_floating();
            else if constexpr (std::is_same_v<ValueClass, int>)
                data_[mapKey] = value.as_integer();
            else if constexpr (std::is_same_v<ValueClass, bool>)
                data_[mapKey] = value.as_boolean();
            else
                data_[mapKey].deserialise(value);
        });
```

Reply:
I was all for this change as it would remove the ugly `toml::find` call, but alas because `toMap()` takes a `const` node pointer, using the nice map-style `node["map"]` lookup isn't possible.  So, leaving as-is.